enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+Internally, the reporting-operator needs to avoid creating `HiveTable` and `PrestoTable` custom resources and instead favor the creation of a `PostgresTable` custom resource when reconciling `ReportDataSource` and `Report` objects that don't have a database table created as of yet. On a similar note, the proper owner references will need to be injected for all `PostgresTable` custom resources to establish a hierarchy. In the case a user deletes a `Report` (or `ReportDataSource`) object, the Kubernetes garbage collector is responsible for cleaning up the correct `PostgresTable` object, and thus requires no further manual intervention from the user.
+
+#### Extending the `StorageLocation` CRD to Support Postgresql
+
+A `StorageLocation` custom resource is responsible for abstracting away details about the underlying metering storage backend, exposing the minimal configuration required to show the end-user where metering data has been persisted. Currently, this object only supports Hive databases and therefore needs to be able to replicate the same functionality we provide with Hive, but with Postgresql interfaces.
+
+When the metering slim installation has been enabled, the default `StorageLocation` custom resource that gets reconciled by the metering-ansible-operator should attempt to make this `CREATE DATABASE metering ...` database call using the Postgres go-client, and update it's status once that database has been created.
+
+Note: Postgresql has no native support for creating a database if it doesn't already exist, so the following call will need to be made instead:
+
+```SQL
+SELECT 'CREATE DATABASE metering' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = 'metering')
+```
+
+#### Outlining a Potential Migration Path
+
+There's currently an undocumented PUSH API for the reporting-operator to statically inject Prometheus data that matches a specific JSON format. This API is largely used in metering end-to-end testing scenarios and is likely, not robust or stable enough to rely on for a full data set migration from existing metering installations. In theory, a migration effort is technically possible, but a more robust API implementation would be needed.
+
+### Risks and Mitigations
+
+- Need to support TLS and authentication for communication between the reporting-operator and the Postgres database instance.
+- Users wanting to use an existing Postgres database instance for metering usage.
+
+## Design Details
+
+### Test Plan
+
+- TBD
```

Comment:
I can think of one good part of a test plan: Implement e2e tests with our more realistic cluster metric data set, https://github.com/kube-reporting/llc-testing-data, and confirm the values in reports come out the same with slim vs. non-slim (current).